### PR TITLE
Add composable node launch actions

### PR DESCRIPTION
**Pull request by hidmic**
This pull request adds `ComposableNodeContainer` and `LoadComposableNodes` actions to `launch_ros`.

Connected to https://github.com/ros2/launch_ros/issues/6. Copy of https://github.com/ros2/launch/pull/197.

### CODE REVIEW

File: launch_ros/launch_ros/actions/composable_node_container.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ) -> None:
+        """
+        Construct a ComposableNodeContainer action.
+
+        Most arguments are forwarded to :class:`launch_ros.actions.Node`, so see
+        the documentation of that class for further details.
+
+        :param composable_node_descriptions: optional descriptions of composable nodes to be loaded
+        """
+        super(ComposableNodeContainer, self).__init__(**kwargs)
```
Comment (sloretz) [inline launch_ros/launch_ros/actions/composable_node_container.py]: nit, `super().__init__(**kwargs)` in python 3

File: launch_ros/launch_ros/actions/load_composable_nodes.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        The container node is expected to provide a `~/_container/load_node` service for
+        loading purposes.
+        Loading will be performed sequentially.
+
+        :param composable_node_descriptions: descriptions of composable nodes to be loaded
+        :param target_container: the container to load the nodes into
+        """
+        self.__composable_node_descriptions = composable_node_descriptions
+        self.__target_container = target_container
+        self.__logger = launch.logging.get_logger(__name__)
```
Comment (sloretz) [inline launch_ros/launch_ros/actions/load_composable_nodes.py]: missing call to `super().__init__(**kwargs)`. Without this I see

```
launch_service.py          355 ERROR    Caught exception in launch (see debug for traceback): 'LoadComposableNodes' object has no attribute '_Action__condition'
```

File: launch_ros/launch_ros/actions/load_composable_nodes.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+    def execute(
+        self,
+        context: LaunchContext
+    ) -> Optional[List[Action]]:
+        """Execute the action."""
+        # Create a client to load nodes in the target container.
+        self.__rclpy_load_node_client = context.locals.launch_ros_node.create_client(
+            composition_interfaces.srv.LoadNode, '{}/_container/load_node'.format(
+                self.__target_container.node_name
```
Comment (sloretz) [inline launch_ros/launch_ros/actions/load_composable_nodes.py]: I think `~/_container/load_node` will expand to `/name/space/node_name/_container/load_node`
This means the `ComposableNodeContainer` must know the node name and namespace. Right now a `ComposableNodeContainer` can be created without specifying these, resulting in `/<node_name_unspecified>/_container/load_node` by default.  

https://github.com/ros2/launch_ros/blob/1340fc8139749b570a43109561c36623c16cbb58/launch_ros/launch_ros/actions/node.py#L171-L172

This isn't a valid service name, so it fails with

```
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Invalid service name: topic name must not contain characters other than alphanumerics, '_', '~', '{', or '}':
```

One solution might be to make `node_name` and `node_namespace` mandatory arguments to `ComposableNodeContainer`'s `__init__()`.

Comment (hidmic) [reply launch_ros/launch_ros/actions/load_composable_nodes.py]: That's a **very** good point. Fixed as suggested.

File: launch_ros/launch_ros/actions/load_composable_nodes.py (hunk 2 above)
Comment (hidmic) [reply launch_ros/launch_ros/actions/load_composable_nodes.py]: Oops. Fixed.

File: launch_ros/launch_ros/actions/composable_node_container.py (hunk 1 above)
Comment (hidmic) [reply launch_ros/launch_ros/actions/composable_node_container.py]: That's true, python2 reflexes. Fixed.

**Comment by hidmic** [comment]
Tested component loading using @mjcarroll draft PR https://github.com/ros2/rclcpp/pull/665 and these launch actions are working as expected.

**Comment by hidmic** [comment]
@sloretz @mjcarroll is there anything else we want to get in before merging this?

**Comment by mjcarroll** [comment]
Nope, good to go from me

**Comment by hidmic** [comment]
CI (full run):

* Linux [Build Status](https://ci.ros2.org/job/ci_linux/6705/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3020/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5509/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6535/)

**Comment by hidmic** [comment]
Going in!

**Pull request merged**